modules/apps/product-navigation/product-navigation-product-menu-api/src/test/java/com/liferay/product/navigation/product/menu/display/context/ProductMenuDisplayContextTest.java

```diff
@@ -0,0 +1,231 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.product.navigation.product.menu.display.context;
+
+import com.liferay.application.list.PanelApp;
+import com.liferay.application.list.PanelCategory;
+import com.liferay.application.list.constants.ApplicationListWebKeys;
+import com.liferay.application.list.constants.PanelCategoryKeys;
+import com.liferay.application.list.display.context.logic.PanelCategoryHelper;
+import com.liferay.portal.configuration.module.configuration.ConfigurationProviderUtil;
+import com.liferay.portal.kernel.servlet.PortletServlet;
+import com.liferay.portal.kernel.test.portlet.MockLiferayResourceRequest;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.JavaConstants;
+import com.liferay.portal.kernel.util.PortalUtil;
+import com.liferay.portal.kernel.util.WebKeys;
+import com.liferay.portal.test.rule.LiferayUnitTestRule;
+import com.liferay.product.navigation.applications.menu.configuration.ApplicationsMenuInstanceConfiguration;
+
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Consumer;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+import org.mockito.ArgumentMatchers;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+
+import org.springframework.mock.web.MockHttpServletRequest;
+
+/**
+ * @author Anderson Luiz
+ */
+public class ProductMenuDisplayContextTest {
+
+	@ClassRule
+	@Rule
+	public static final LiferayUnitTestRule liferayUnitTestRule =
+		LiferayUnitTestRule.INSTANCE;
+
+	@Before
+	public void setUp() throws Exception {
+		_themeDisplay = Mockito.mock(ThemeDisplay.class);
+
+		_mockHttpServletRequest = new MockHttpServletRequest();
+
+		_mockPortletRequest = new MockLiferayResourceRequest();
+
+		_mockPortletRequest.setAttribute(
+			PortletServlet.PORTLET_SERVLET_REQUEST, _mockHttpServletRequest);
+
+		_mockPortletRequest.setAttribute(WebKeys.THEME_DISPLAY, _themeDisplay);
+
+		_mockHttpServletRequest.setAttribute(
```

Comment:
Brian will probably say to put these lines _before_ line 57 `_mockPortletRequest = new MockLiferayResourceRequest();`

Reply:
I believe the reason is [this convention](https://liferay.atlassian.net/wiki/spaces/ENGR/pages/1511752519/Java+conventions+in+Liferay#%5BinlineExtension%5DVariables-are-only-declared-as-used%2C-not-before-(%E2%80%9CAs-used%E2%80%9D))

Reply:
It's basically, create it, use it, read it/from it. Will work for almost every case